src/ServiceClient/Implements/SocketDataTransfer.cs

```diff
@@ -0,0 +1,34 @@
+﻿using ServiceClient.Abstractions;
+using System.Net.WebSockets;
+using System.Text;
+
+namespace ServiceClient.Implements
+{
+    public class SocketDataTransfer : ISocketDataTransfer
+    {
+        private const int READ_BUFFER_SIZE = 4096;
+
+        public async Task<string> ReceiveAsync(ClientWebSocket socket, CancellationToken cancellationToken)
+        {
+            StringBuilder sb = new StringBuilder();
+            var buffer = new byte[READ_BUFFER_SIZE];
```

Comment:
I think the _StringBuilder_ and the _buffer_ should be promoted as global states in the class to avoid creating garbage on the heap.

Inside the method instead doing

`sb = new StringBuilder() `

use

`sb.Clear()`

This method will be used in a hot path!

Reply:
OK